<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Microbiome Analysis Tools for R • micro4R</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png">
<link rel="icon" sizes="any" href="favicon.ico">
<link rel="manifest" href="site.webmanifest">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Microbiome Analysis Tools for R">
<meta name="description" content="Provides a suite of convenience functions for microbiome analysis.">
<meta property="og:description" content="Provides a suite of convenience functions for microbiome analysis.">
<meta property="og:image" content="https://mshilts1.github.io/micro4R/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">micro4R</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="micro4r">micro4R<a class="anchor" aria-label="anchor" href="#micro4r"></a>
</h1>
</div>
<p>Microbiome data analysis tools for R</p>
<!-- badges: start -->

<p>The goal of <code>micro4R</code> was to create an R package for microbiome data processing with a low barrier to entry. I started my career in microbiome research at the bench and had to <a href="https://media0.giphy.com/media/v1.Y2lkPTc5MGI3NjExY3hrYzg1a2I2eGtuNWIwYTRqNDMzNGE0cWlkNGE5OXB4ZHV1YXY4dCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/WsNbxuFkLi3IuGI9NU/giphy.gif" class="external-link">ELI5</a> to myself how to process and analyze “big data”. I’ve spent a ton of time poring over and experimenting with <a href="https://github.com/mshilts1/micro4R?tab=readme-ov-file#acknowledgements" class="external-link">others’ code</a>, and want to pass it on. <a href="https://mshilts1.github.io/micro4R/"><img src="reference/figures/logo.png" align="right" height="139" alt="micro4R website"></a></p>
<p>Likely, the ideal person to benefit from <code>micro4R</code> would be a bench scientist without much formal statistics or bioinformatics training. Fair warning, if you already have a strong stats/informatics background, this may not be of much use for you!</p>
<p>This package does not create any brand new functionality and is essentially inspired by and/or a wrapper of existing tools <a href="https://github.com/mshilts1/micro4R?tab=readme-ov-file#acknowledgements" class="external-link">others</a> have already created. Much of what it does can be accomplished with other packages, such as <a href="https://bioconductor.org/packages/release/bioc/html/phyloseq.html" class="external-link">phyloseq</a>, <a href="https://qiime2.org" class="external-link">QIIME 2</a>, and <a href="https://www.microbiomeanalyst.ca" class="external-link">MicrobiomeAnalyst</a>. One of these may be better for your purposes, and I’d encourage anyone new to the field to explore multiple tools!</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>All you really need is <a href="https://cran.rstudio.com" class="external-link">R</a>, but I’d recommend also downloading and working in <a href="https://posit.co/download/rstudio-desktop/" class="external-link">RStudio</a>. If you’re a true newbie to R, there’s tons of free <a href="https://www.reddit.com/user/jjkraker/comments/zfhe1e/i_want_to_learn_basics_of_r_if_so_heres_a_reading/" class="external-link">content</a> to help you learn the basics.</p>
<p>Once you’re set in R/RStudio, you can install the development version of micro4R like so:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("pak")</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pak.html" class="external-link">pak</a></span><span class="op">(</span><span class="st">"mshilts1/micro4R"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="my-usual-workflow">My Usual Workflow<a class="anchor" aria-label="anchor" href="#my-usual-workflow"></a>
</h2>
<p>I’ve listed the steps that I usually take with 16S data, some of which will be demonstrated below.</p>
<ol style="list-style-type: decimal">
<li>Do the lab work to extract DNA, make libraries, submit for sequencing, etc. <a href="https://github.com/mshilts1/micro4R?tab=readme-ov-file#step-1-the-lab-work" class="external-link">Go ↓</a><br>
</li>
<li>Get the data from the sequencing core. <a href="https://github.com/mshilts1/micro4R?tab=readme-ov-file#step-2-sequencing" class="external-link">Go ↓</a><br>
</li>
<li>Process the data through <a href="https://benjjneb.github.io/dada2/" class="external-link">dada2</a> then <a href="https://benjjneb.github.io/decontam/vignettes/decontam_intro.html" class="external-link">decontam</a> to generate a cleaned <a href="https://www.nature.com/articles/nmeth.3869" class="external-link">ASV (amplicon sequencing variant) count table and its associated taxonomy table</a>. <a href="https://github.com/mshilts1/micro4R?tab=readme-ov-file#step-3-data-processing" class="external-link">Go ↓</a><br>
</li>
<li>Create a “metadata” file with pertinent information on the samples and controls in your run. <a href="https://github.com/mshilts1/micro4R?tab=readme-ov-file#step-4-metadata" class="external-link">Go ↓</a>
</li>
<li>Bioinformatically remove suspected contaminants <a href="">Go ↓</a>.<br>
</li>
<li>Do some basic sanity checking on the metadata, ASV, and taxonomy objects. <a href="">Go ↓</a><br>
</li>
<li>Check the quality of the sequencing data by examining both the positive and negative controls, and additionally how the controls compare to your real samples. <a href="">Go ↓</a><br>
</li>
<li>Alpha diversity (vegan).<br>
</li>
<li>Beta diversity (vegan).<br>
</li>
<li>Differential abundance (maaslin3).</li>
</ol>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>I’ll run through a very small and simple possible use case below. For more detailed help and documentation, please explore the vignettes <em>(TBA)</em>.</p>
<div class="section level3">
<h3 id="step-1-the-lab-work">Step 1: the lab work<a class="anchor" aria-label="anchor" href="#step-1-the-lab-work"></a>
</h3>
<p>I’m not going to get into much detail here. More TBA. Both negative and positive controls are very important! Why, and what are those? Read more here: <sup><a href="https://pubmed.ncbi.nlm.nih.gov/25387460/" class="external-link">1</a>,</sup><sup><a href="https://pubmed.ncbi.nlm.nih.gov/27239228/" class="external-link">2</a>,</sup><sup><a href="https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-020-01839-y" class="external-link">3</a>,</sup><sup><a href="https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0738-z" class="external-link">4</a>,</sup><sup><a href="https://journals.asm.org/doi/10.1128/msystems.00062-16" class="external-link">5</a></sup></p>
</div>
<div class="section level3">
<h3 id="step-2-sequencing">Step 2: Sequencing<a class="anchor" aria-label="anchor" href="#step-2-sequencing"></a>
</h3>
<p>You’ll need to follow your own desired protocol to perform the sequencing, and get the data back.</p>
<p>Included with the package is an extremely and unnaturally tiny toy example to demonstrate its major functionality, using subsampled publicly available nasal swab 16S microbiome sequencing <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA726992" class="external-link">data</a> that I generated along with many <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC8819187/" class="external-link">colleagues</a>.</p>
</div>
<div class="section level3">
<h3 id="step-3-data-processing">Step 3: Data processing<a class="anchor" aria-label="anchor" href="#step-3-data-processing"></a>
</h3>
<p>The files received from the sequencer will most likely be FASTQ files. These are files with DNA sequences generated by the sequencer along with quality score information (what the Q in FASTQ is for). In the age of next-generation sequencing, you could easily get millions of these sequences per each sequencing run. Our poor human brains can’t really comprehend or do anything with this data as it is. So what we’re going to do is process this data to eventually turn it into two tables that we humans can make sense of and analyze.</p>
<p>We’re going to use the R package <code>dada2</code> to turn the FASTQ files into matching amplicon sequence variant (ASV) count and taxonomy tables. If you don’t know what an ASV is, please go <a href="https://benjjneb.github.io/dada2/index.html" class="external-link">here</a> first.</p>
<p>The first thing we’ll do on these files is run <code><a href="reference/dada2_asvtable.html">dada2_asvtable()</a></code>, which is essentially a wrapper to generate an ASV count table by following a workflow similar to the <a href="https://benjjneb.github.io/dada2/tutorial.html" class="external-link">dada2 tutorial</a>.</p>
<p>This function can take a number of arguments, but the most important one is ‘where’, which is the path to the folder where your FASTQ files are located.</p>
<p>For demonstration purposes, it’s been set to the relative path of the example FASTQ files that are included with the package:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mshilts1.github.io/micro4R/">micro4R</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; This is version 0.0.0.9000 of micro4R. CAUTION: This is package is under active development and its functions may change at any time, without warning! Please visit https://github.com/mshilts1/micro4R to see recent changes.</span></span>
<span></span>
<span><span class="va">asvtable</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/dada2_asvtable.html">dada2_asvtable</a></span><span class="op">(</span>where <span class="op">=</span> <span class="st">"inst/extdata/f"</span>, chatty <span class="op">=</span> <span class="cn">FALSE</span>, logfile <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Creating output directory: /var/folders/pp/15rq6p297j18gk2xt39kdmm40000gp/T//Rtmpi2dRcG/dada2_out/filtered</span></span>
<span><span class="co">#&gt; 59520 total bases in 248 reads from 7 samples will be used for learning the error rates.</span></span>
<span><span class="co">#&gt; 49600 total bases in 248 reads from 7 samples will be used for learning the error rates.</span></span></code></pre></div>
<p>If you’re running this with your own data, set ‘where’ to the path of the folder where your FASTQ files are stored. If you leave it empty (e.g., run <code><a href="reference/dada2_asvtable.html">dada2_asvtable()</a></code>), it will default to searching in your current working directory. (‘chatty’ was set to FALSE because tons of information gets printed to the console otherwise; I’d recommend setting it to TRUE (the default) when you’re processing data for real, as the information is useful, but just too much here.)</p>
<p>Let’s take a quick look at what this asvtable looks like (using the <code><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">tibble::as_tibble()</a></code> function so it prints more nicely):</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">asvtable</span>, rownames <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 7 × 7</span></span>
<span><span class="co">#&gt;   SampleID  TACGTAGGTGGCAAGCGTTA…¹ TACGGAGGGTGCAAGCGTTA…² TACGTAGGGTGCGAGCGTTG…³</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;                      &lt;int&gt;                  &lt;int&gt;                  &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1 SAMPLED_…                      0                      0                      0</span></span>
<span><span class="co">#&gt; 2 SAMPLED_…                      0                      0                      0</span></span>
<span><span class="co">#&gt; 3 SAMPLED_…                     44                      0                      0</span></span>
<span><span class="co">#&gt; 4 SAMPLED_…                     24                      0                      0</span></span>
<span><span class="co">#&gt; 5 SAMPLED_…                      0                      0                     12</span></span>
<span><span class="co">#&gt; 6 SAMPLED_…                      0                     35                      6</span></span>
<span><span class="co">#&gt; 7 SAMPLED_…                      0                      0                      0</span></span>
<span><span class="co">#&gt; # ℹ abbreviated names:</span></span>
<span><span class="co">#&gt; #   ¹​TACGTAGGTGGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGCGCGTAGGCGGTTTTTTAAGTCTGATGTGAAAGCCCACGGCTCAACCGTGGAGGGTCATTGGAAACTGGAAAACTTGAGTGCAGAAGAGGAAAGTGGAATTCCATGTGTAGCGGTGAAATGCGCAGAGATATGGAGGAACACCAGTGGCGAAGGCGACTTTCTGGTCTGTAACTGACGCTGATGTGCGAAAGCGTGGGGATCAAACAGG,</span></span>
<span><span class="co">#&gt; #   ²​TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTCTGTCAAGTCGGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATTCGAAACTGGCAGGCTAGAGTCTTGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACAAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGG,</span></span>
<span><span class="co">#&gt; #   ³​TACGTAGGGTGCGAGCGTTGTCCGGAATTACTGGGCGTAAAGGGCTCGTAGGTGGTTTGTCGCGTCGTCTGTGAAATTCTGGGGCTTAACTCCGGGCGTGCAGGCGATACGGGCATAACTTGAGTGCTGTAGGGGTAACTGGAATTCCTGGTGTAGCGGTGAAATGCGCAGATATCAGGAGGAACACCGATGGCGAAGGCAGGTTACTGGGCAGTTACTGACGCTGAGGAGCGAAAGCATGGGTAGCGAACAGG</span></span>
<span><span class="co">#&gt; # ℹ 3 more variables:</span></span>
<span><span class="co">#&gt; #   TACGTAGGGTGCAAGCGTTGTCCGGAATTACTGGGCGTAAAGAGCTCGTAGGTGGTTTGTCACGTCGTCTGTGAAATTCCACAGCTTAACTGTGGGCGTGCAGGCGATACGGGCTGACTTGAGTACTGTAGGGGTAACTGGAATTCCTGGTGTAGCGGTGAAATGCGCAGATATCAGGAGGAACACCGATGGCGAAGGCAGGTTACTGGGCAGTTACTGACGCTGAGGAGCGAAAGCATGGGTAGCAAACAGG &lt;int&gt;,</span></span>
<span><span class="co">#&gt; #   TACGTAGGTGACAAGCGTTGTCCGGATTTATTGGGCGTAAAGGGAGCGCAGGCGGTCTGTTTAGTCTAATGTGAAAGCCCACGGCTTAACCGTGGAACGGCATTGGAAACTGACAGACTTGAATGTAGAAGAGGAAAATGGAATTCCAAGTGTAGCGGTGGAATGCGTAGATATTTGGAGGAACACCAGTGGCGAAGGCGATTTTCTGGTCTAACATTGACGCTGAGGCTCGAAAGCGTGGGGAGCGAACAGG &lt;int&gt;, …</span></span></code></pre></div>
<p>This is basically just a count table of the number of ASVs detected in each sample.</p>
<p>Let’s look what the column names (AKA the names of the ASVs) look like:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">asvtable</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "TACGTAGGTGGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGCGCGTAGGCGGTTTTTTAAGTCTGATGTGAAAGCCCACGGCTCAACCGTGGAGGGTCATTGGAAACTGGAAAACTTGAGTGCAGAAGAGGAAAGTGGAATTCCATGTGTAGCGGTGAAATGCGCAGAGATATGGAGGAACACCAGTGGCGAAGGCGACTTTCTGGTCTGTAACTGACGCTGATGTGCGAAAGCGTGGGGATCAAACAGG" </span></span>
<span><span class="co">#&gt; [2] "TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTCTGTCAAGTCGGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATTCGAAACTGGCAGGCTAGAGTCTTGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACAAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGG" </span></span>
<span><span class="co">#&gt; [3] "TACGTAGGGTGCGAGCGTTGTCCGGAATTACTGGGCGTAAAGGGCTCGTAGGTGGTTTGTCGCGTCGTCTGTGAAATTCTGGGGCTTAACTCCGGGCGTGCAGGCGATACGGGCATAACTTGAGTGCTGTAGGGGTAACTGGAATTCCTGGTGTAGCGGTGAAATGCGCAGATATCAGGAGGAACACCGATGGCGAAGGCAGGTTACTGGGCAGTTACTGACGCTGAGGAGCGAAAGCATGGGTAGCGAACAGG"</span></span>
<span><span class="co">#&gt; [4] "TACGTAGGGTGCAAGCGTTGTCCGGAATTACTGGGCGTAAAGAGCTCGTAGGTGGTTTGTCACGTCGTCTGTGAAATTCCACAGCTTAACTGTGGGCGTGCAGGCGATACGGGCTGACTTGAGTACTGTAGGGGTAACTGGAATTCCTGGTGTAGCGGTGAAATGCGCAGATATCAGGAGGAACACCGATGGCGAAGGCAGGTTACTGGGCAGTTACTGACGCTGAGGAGCGAAAGCATGGGTAGCAAACAGG" </span></span>
<span><span class="co">#&gt; [5] "TACGTAGGTGACAAGCGTTGTCCGGATTTATTGGGCGTAAAGGGAGCGCAGGCGGTCTGTTTAGTCTAATGTGAAAGCCCACGGCTTAACCGTGGAACGGCATTGGAAACTGACAGACTTGAATGTAGAAGAGGAAAATGGAATTCCAAGTGTAGCGGTGGAATGCGTAGATATTTGGAGGAACACCAGTGGCGAAGGCGATTTTCTGGTCTAACATTGACGCTGAGGCTCGAAAGCGTGGGGAGCGAACAGG" </span></span>
<span><span class="co">#&gt; [6] "TACGTAGGTCCCGAGCGTTGTCCGGATTTATTGGGCGTAAAGCGAGCGCAGGCGGTTAGATAAGTCTGAAGTTAAAGGCTGTGGCTTAACCATAGTACGCTTTGGAAACTGTTTAACTTGAGTGCAAGAGGGGAGAGTGGAATTCCATGTGTAGCGGTGAAATGCGTAGATATATGGAGGAACACCGGTGGCGAAAGCGGCTCTCTGGCTTGTAACTGACGCTGAGGCTCGAAAGCGTGGGGAGCAAACAGG"</span></span></code></pre></div>
<p>Our ASVs are by default just named after their literal DNA sequences. Since you’re (probably?) not a computer, a string of hundreds of nucletotides is likely not something you can make much sense of by yourself. The next step will take those nucleotide sequences and compare them against a database (or two) of sequences with known taxonomy:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">train</span> <span class="op">&lt;-</span> <span class="st">"inst/extdata/db/EXAMPLE_silva_nr99_v138.2_toGenus_trainset.fa.gz"</span> <span class="co"># set training database</span></span>
<span><span class="va">species</span> <span class="op">&lt;-</span> <span class="st">"inst/extdata/db/EXAMPLE_silva_v138.2_assignSpecies.fa.gz"</span> <span class="co"># set species database</span></span>
<span></span>
<span><span class="va">taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/dada2_taxa.html">dada2_taxa</a></span><span class="op">(</span>asvtable <span class="op">=</span> <span class="va">asvtable</span>, train <span class="op">=</span> <span class="va">train</span>, species <span class="op">=</span> <span class="va">species</span>, chatty <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>There are two databases that we’re using for taxonomic assignment here:<br>
1. ‘train’ needs to be the path to whatever database you’d like to use as the “training set of reference sequences with known taxonomy”.<br>
2. ‘species’ is OPTIONAL. If you’d like to use this option, provide the path to a specifically formatted species assignment database. (Read more <a href="https://benjjneb.github.io/dada2/assign.html#species-assignment" class="external-link">here</a>.)</p>
<p><strong>CAUTION: The two databases used in the example here are comically small and artificial subsamples of the real SILVA databases, and should only ever be used for testing and demonstration purposes!</strong> You’ll definitely need to download and use the real <a href="https://benjjneb.github.io/dada2/training.html" class="external-link">databases</a> for your actual data!</p>
<p>There are many options for taxonomic databases you can use; the major players are SILVA, RDP, GreenGenes, and UNITE. Please go <a href="https://benjjneb.github.io/dada2/training.html" class="external-link">here</a> for details and links. I usually prefer the SILVA databases, but you don’t have to!</p>
<p>Let’s take a look at the taxonomy assignment table:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 × 7</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co">#&gt;   ASV                                    Kingdom Phylum Class Order Family Genus</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                                  &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#&gt; 1 TACGTAGGTGGCAAGCGTTATCCGGAATTATTGGGCG… Bacter… Bacil… Baci… Stap… Staph… Stap…</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt; 2 TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCG… Bacter… Pseud… Gamm… Ente… Enter… Kleb…</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; 3 TACGTAGGGTGCGAGCGTTGTCCGGAATTACTGGGCG… Bacter… Actin… Acti… Myco… Coryn… Cory…</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; 4 TACGTAGGGTGCAAGCGTTGTCCGGAATTACTGGGCG… Bacter… Actin… Acti… Myco… Coryn… Cory…</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; 5 TACGTAGGTGACAAGCGTTGTCCGGATTTATTGGGCG… Bacter… Bacil… Baci… Lact… Carno… Dolo…</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; 6 TACGTAGGTCCCGAGCGTTGTCCGGATTTATTGGGCG… Bacter… Bacil… Baci… Lact… Strep… Stre…</span></span></code></pre></div>
<p>It’s a bit squished, but you can see this information is more human-friendly here. Each ASV has been given a taxonomic assignment the lowest taxonomic level the taxonomy assigner was confident of.</p>
</div>
<div class="section level3">
<h3 id="step-4-metadata">Step 4: Metadata<a class="anchor" aria-label="anchor" href="#step-4-metadata"></a>
</h3>
<p>Next, we need to load in some metadata about our samples.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/example_metadata.html">example_metadata</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metadata</span></span>
<span><span class="co">#&gt; # A tibble: 7 × 7</span></span>
<span><span class="co">#&gt;   SampleID                 LabID SampleType host_age host_sex Host_disease neg  </span></span>
<span><span class="co">#&gt;   &lt;chr&gt;                    &lt;chr&gt; &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;    &lt;chr&gt;        &lt;lgl&gt;</span></span>
<span><span class="co">#&gt; 1 SAMPLED_5080-MS-1_328-G… part… Homo sapi…       33 female   healthy      FALSE</span></span>
<span><span class="co">#&gt; 2 SAMPLED_5080-MS-1_339-A… part… Homo sapi…       25 male     healthy      FALSE</span></span>
<span><span class="co">#&gt; 3 SAMPLED_5348-MS-1_162-A… part… Homo sapi…       27 male     COVID-19     FALSE</span></span>
<span><span class="co">#&gt; 4 SAMPLED_5348-MS-1_297-G… part… Homo sapi…       26 female   COVID-19     FALSE</span></span>
<span><span class="co">#&gt; 5 SAMPLED_5080-MS-1_307-A… CTRL… negative …       NA &lt;NA&gt;     &lt;NA&gt;         TRUE </span></span>
<span><span class="co">#&gt; 6 SAMPLED_5080-MS-1_313-G… CTRL… negative …       NA &lt;NA&gt;     &lt;NA&gt;         TRUE </span></span>
<span><span class="co">#&gt; 7 SAMPLED_5348-MS-1_381-T… CTRL… positive …       NA &lt;NA&gt;     &lt;NA&gt;         FALSE</span></span></code></pre></div>
<p>Let’s look at the ‘SampleID’ field, which is what is sound like, and uniquely identifies each sample:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span><span class="op">$</span><span class="va">SampleID</span></span>
<span><span class="co">#&gt; [1] "SAMPLED_5080-MS-1_328-GATCTACG-TCGACGAG_S328_L001"</span></span>
<span><span class="co">#&gt; [2] "SAMPLED_5080-MS-1_339-ACTCACTG-GATCGTGT_S339_L001"</span></span>
<span><span class="co">#&gt; [3] "SAMPLED_5348-MS-1_162-ACGTGCGC-GGATATCT_S162_L001"</span></span>
<span><span class="co">#&gt; [4] "SAMPLED_5348-MS-1_297-GTCTGCTA-ACGTCTCG_S297_L001"</span></span>
<span><span class="co">#&gt; [5] "SAMPLED_5080-MS-1_307-ATAGTACC-ACGTCTCG_S307_L001"</span></span>
<span><span class="co">#&gt; [6] "SAMPLED_5080-MS-1_313-GACATAGT-TCGACGAG_S313_L001"</span></span>
<span><span class="co">#&gt; [7] "SAMPLED_5348-MS-1_381-TGCTCGTA-GTCAGATA_S381_L001"</span></span></code></pre></div>
<p>The first thing you may notice is the ‘SampleIDs’ here are the kinds of IDs that only a computer could love. For my standard workflow, I like to keep the SampleIDs as the FASTQ file names because they will by default automatically match the SampleIDs generated with dada2_asvtable().</p>
<p>In this example, there is also a ‘LabID’ field, which is an ID that could have been used all through specimen processing, as it is much more human-friendly:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span><span class="op">$</span><span class="va">LabID</span></span>
<span><span class="co">#&gt; [1] "participant01" "participant02" "participant03" "participant04"</span></span>
<span><span class="co">#&gt; [5] "CTRL_neg_ext"  "CTRL_neg_pcr"  "CTRL_pos_pcr"</span></span></code></pre></div>
<p>To seemlessly use this package, you MUST have a column called ‘SampleID’, and those IDs must exactly match between your metadata and ASV count table objects. But otherwise, you’re free to name your samples whatever you want.</p>
</div>
<div class="section level3">
<h3 id="step-5-bioinformatic-decontamination">Step 5: Bioinformatic Decontamination<a class="anchor" aria-label="anchor" href="#step-5-bioinformatic-decontamination"></a>
</h3>
<p>What kind of information you’ll need to have in your metadata object is HIGHLY dependent on your study, but there’s some information that we must have for the optional (but highly recommended!) processing of your ASV table through <a href="https://github.com/benjjneb/decontam" class="external-link">decontam</a>.</p>
<p>What does <code>decontam</code> need to know?<br>
1. To use the “prevalence” method: which samples are the negative controls.<br>
2. To use the “frequency” method: the DNA concentration in each sample prior to sequencing.<br>
3. To use both/either of the “prevalence” and “frequency” methods, you need all the above.</p>
<p>Don’t have any negative controls? You won’t be able to run <code>decontam</code>, and I strongly recommend you include some next time! Both negative and positive controls are very important! Read more here: <sup><a href="https://pubmed.ncbi.nlm.nih.gov/25387460/" class="external-link">1</a>,</sup><sup><a href="https://pubmed.ncbi.nlm.nih.gov/27239228/" class="external-link">2</a>,</sup><sup><a href="https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-020-01839-y" class="external-link">3</a>,</sup><sup><a href="https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0738-z" class="external-link">4</a>,</sup><sup><a href="https://journals.asm.org/doi/10.1128/msystems.00062-16" class="external-link">5</a></sup></p>
<p>In our example data, we only are able to use the “prevalence” method, because we know which samples were negative controls, and which were not. That information is in our column called “neg”, where TRUE means it was a negative control:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span><span class="op">$</span><span class="va">neg</span></span>
<span><span class="co">#&gt; [1] FALSE FALSE FALSE FALSE  TRUE  TRUE FALSE</span></span></code></pre></div>
<p>Next, let’s run the <code><a href="reference/decontam_wrapper.html">decontam_wrapper()</a></code> on the example data we’ve generated so far:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/decontam_wrapper.html">decontam_wrapper</a></span><span class="op">(</span>asvtable <span class="op">=</span> <span class="va">asvtable</span>, taxa <span class="op">=</span> <span class="va">taxa</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, logfile <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in .is_contaminant(seqtab, conc = conc, neg = neg, method = method, :</span></span>
<span><span class="co">#&gt; Removed 3 samples with zero total counts (or frequency).</span></span>
<span><span class="co">#&gt; Warning in .is_contaminant(seqtab, conc = conc, neg = neg, method = method, :</span></span>
<span><span class="co">#&gt; Some batches have very few (&lt;=4) samples.</span></span>
<span><span class="co">#&gt; Warning in .is_contaminant(seqtab, conc = conc, neg = neg, method = method, :</span></span>
<span><span class="co">#&gt; Removed 3 samples with zero total counts (or frequency).</span></span>
<span><span class="co">#&gt; Warning in .is_contaminant(seqtab, conc = conc, neg = neg, method = method, :</span></span>
<span><span class="co">#&gt; Some batches have very few (&lt;=4) samples.</span></span>
<span><span class="co">#&gt; [1] "No contaminants were detected. Exiting function and returning your original ASV and taxa tables."</span></span></code></pre></div>
<p>You’ll see several messages, including one at the bottom that tells us “No contaminants were detected. Exiting function and returning your original ASV and taxa tables.” With real data with proper controls, you should not expect to ever see this message. Our example data set was just too stupidly small for <code>decontam</code> to work properly!</p>
<p>So that I can demonstrate <code>decontam</code> actually doing something, we’ll deliberately “contaminate” the ASV table with the <code><a href="reference/contaminate.html">contaminate()</a></code> command. All we’re doing is artificially adding counts to one of the ASVs, and throwing in a few more negative controls for good measure. This data is even more fake and made up than it was previously! The command <code><a href="reference/contaminate.html">contaminate()</a></code> will also simultaneously create a matched metadata object with the additional made up negative controls.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contaminated_asvtable</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/converter.html">converter</a></span><span class="op">(</span><span class="fu"><a href="reference/contaminate.html">contaminate</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">asvtable</span><span class="op">)</span></span>
<span><span class="va">contaminated_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/dada2_taxa.html">dada2_taxa</a></span><span class="op">(</span>asvtable <span class="op">=</span> <span class="va">contaminated_asvtable</span>, train <span class="op">=</span> <span class="va">train</span>, species <span class="op">=</span> <span class="va">species</span><span class="op">)</span> <span class="co"># we don't actually have to re-run this command with this specific example, but it's good practice to always ensure your asvtable and taxa tables match.</span></span>
<span><span class="co">#&gt; [1] "CAUTION: You're using the provided micro4R EXAMPLE reference databases. These are extremely tiny and unrealistic and meant only for testing and demonstration purposes. DO NOT use them with your real data."</span></span>
<span><span class="co">#&gt; Finished processing reference fasta.</span></span>
<span><span class="va">contaminated_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/contaminate.html">contaminate</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">metadata</span></span>
<span><span class="va">decontaminated</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/decontam_wrapper.html">decontam_wrapper</a></span><span class="op">(</span>asvtable <span class="op">=</span> <span class="va">contaminated_asvtable</span>, taxa <span class="op">=</span> <span class="va">contaminated_taxa</span>, metadata <span class="op">=</span> <span class="va">contaminated_metadata</span>, logfile <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in .is_contaminant(seqtab, conc = conc, neg = neg, method = method, :</span></span>
<span><span class="co">#&gt; Removed 1 samples with zero total counts (or frequency).</span></span>
<span><span class="co">#&gt; Warning in .is_contaminant(seqtab, conc = conc, neg = neg, method = method, :</span></span>
<span><span class="co">#&gt; Removed 1 samples with zero total counts (or frequency).</span></span></code></pre></div>
<p>The object “decontaminated” contains a list of the… decontaminated… ASV table, taxa table, and for our convenience also includes the metadata table.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">contaminated_taxa</span><span class="op">)</span> <span class="co"># number of rows corresponds to number of ASVs</span></span>
<span><span class="co">#&gt; [1] 6</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">decontaminated</span><span class="op">$</span><span class="va">taxa</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 5</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">contaminated_asvtable</span><span class="op">)</span> <span class="co"># number of columns corresponds to number of ASVs</span></span>
<span><span class="co">#&gt; [1] 6</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">decontaminated</span><span class="op">$</span><span class="va">asvtable</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 5</span></span></code></pre></div>
<p>The decontaminated version of both the ASV and taxa tables have one fewer ASV than the originals, as I artifically added a bunch of counts to one of the ASVs to the contaminated version of the ASV table. <code>decontam</code> rightfully thinks that ASV is likely a background contaminant due to its prevalence in the negative controls, and has now removed it, to make our data cleaner and more reliable. Why is this so important?^[Negative controls are especially important with 16S data due the nature of the process: 1) bacteria and their DNA are ubiquitous and can live even in environments hostile to most other life, 2) the PCR protocol deliberately enriches for all bacteria in a semi-universal way. This means the data can be extremely susceptible to contamination. The details of this are out of the scope of this document, but your FIRST step should be improving your lab methods to reduce contamination potential as much as possible.</p>
<p>However, no matter how amazing you (or your colleagues) are in the lab, you’ll probably still have at least some contamination. That’s where the negative controls come in. I would recommend having at least one negative extraction control (e.g., extract some ultraclean water or sample buffer) per every extraction batch, and a PCR negative control for every PCR master mix batch.]</p>
</div>
<div class="section level3">
<h3 id="step-6-sanity-check">Step 6: Sanity Check<a class="anchor" aria-label="anchor" href="#step-6-sanity-check"></a>
</h3>
</div>
<div class="section level3">
<h3 id="step-7-quality-check">Step 7: Quality Check<a class="anchor" aria-label="anchor" href="#step-7-quality-check"></a>
</h3>
</div>
<div class="section level3">
<h3 id="step-8-alpha-diversity">Step 8: Alpha diversity<a class="anchor" aria-label="anchor" href="#step-8-alpha-diversity"></a>
</h3>
</div>
<div class="section level3">
<h3 id="step-9-beta-diversity">Step 9: Beta diversity<a class="anchor" aria-label="anchor" href="#step-9-beta-diversity"></a>
</h3>
</div>
<div class="section level3">
<h3 id="step-10-additional-analysis">Step 10: Additional analysis<a class="anchor" aria-label="anchor" href="#step-10-additional-analysis"></a>
</h3>
<hr>
<p>Move information to the bottom for anyone who wants more details</p>
<p>subsampled FASTQ files from a <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC8819187/" class="external-link">manuscript</a> I co-authored with my colleagues, for which the raw data is publicly available under bioproject ID <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA726992" class="external-link">PRJNA726992</a>. From seven samples from this study, using <a href="https://github.com/lh3/seqtk" class="external-link">seqtk</a>, I randomly sampled <strong>only 50 reads</strong> from each FASTQ file so that the files would take up minimal space and the example would run quickly.</p>
<p>If you’d like to run through this the full fastq files can be downloaded from SRA or as a zipped bolus <a href="https://drive.google.com/file/d/1NOvmsxFxWb1Vigq8rdb5SCfLLNu-Qjy8/view?usp=sharing" class="external-link">here</a></p>
<p>Here is some body text that needs a footnote.<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;This is the content of the footnote.&lt;/p&gt;"><sup>1</sup></a></p>
<ul>
<li>logo made by me using artwork from <a href="https://www.canva.com/" class="external-link">Canva</a> (©<a href="https://www.canva.com/p/id/BAClqvm1MBE/" class="external-link">iconbunny11</a>) followed by <a href="https://github.com/GuangchuangYu/hexSticker" class="external-link">hexSticker</a> to get it into the typical hex logo format.</li>
</ul>
<p>Not going to keep this on the readme, but want to hold onto the logo code until I put it somewhere else.<br>
s &lt;- sticker(image_path, package=“micro4R”, p_size=15, p_family = “Comfortaa”, p_fontface = “bold”, p_y = 1.5, s_x=1, s_y=.75, s_width=.5, s_height = .5, p_color = “black”, h_fill = “#6ed5f5”, h_color= “#16bc93”, h_size = 2, filename=“inst/figures/imgfile.png”).</p>
<p>built R 4.5.1<br>
RStudio Version 2025.05.1+513 (2025.05.1+513) macOS Sequoia Version 15.6.1</p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h1>
<p>As mentioned above, I could not have done any of this without the benefit of what the work of others. These tools below were especially important, as they were either extremely influential in helping me learn how to do these analyses, and/or are directly used here:</p>
<ul>
<li>
<a href="https://mothur.org" class="external-link">mothur</a><br>
</li>
<li>
<a href="https://qiime2.org" class="external-link">qiime</a><br>
</li>
<li>
<a href="https://github.com/andreyto/mgsat" class="external-link">mgsat</a><br>
</li>
<li>maaslin<a href="https://huttenhower.sph.harvard.edu/maaslin/" class="external-link">2</a>/<a href="https://huttenhower.sph.harvard.edu/maaslin3/" class="external-link">3</a><br>
</li>
<li><a href="https://cran.r-project.org/web/packages/vegan/index.html" class="external-link">vegan</a></li>
<li>
<a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a>, especially <a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a> and <a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a>. In addition to these packages, Hadley Wickham and team have written several extremely helpful <a href="https://hadley.nz" class="external-link">guides and tutorials</a> on data science.</li>
<li><a href="https://callahanlab.cvm.ncsu.edu/software/" class="external-link">dada2 and decontam</a></li>
<li>
<a href="https://hbiostat.org" class="external-link">Suite from Dr. Frank Harrell</a>, especially <a href="https://cran.r-project.org/web/packages/rms/index.html" class="external-link">rms</a> and <a href="https://cran.r-project.org/web/packages/Hmisc/index.html" class="external-link">Hmisc</a>.</li>
<li>Colleague and physician-scientist <a href="https://pediatrics.vumc.org/person/christian-rosas-salazar-md-mph" class="external-link">Dr. Christian Rosas-Salazar</a> is talented at many things, but has an especial knack for creating figures that are both beautiful and informative.</li>
</ul>
<p>More acknowledgements and more details to be added later</p>
</div>


  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing micro4R</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Meghan Shilts <br><small class="roles"> Author, maintainer </small>   </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://CRAN.R-project.org/" class="external-link"><img src="https://www.r-pkg.org/badges/version/micro4R" alt="CRAN status"></a></li>
<li><a href="https://app.codecov.io/gh/mshilts1/micro4R" class="external-link"><img src="https://codecov.io/gh/mshilts1/micro4R/graph/badge.svg" alt="Codecov test coverage"></a></li>
<li><a href="https://github.com/mshilts1/micro4R/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/mshilts1/micro4R/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Meghan Shilts.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
