<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Wrapper of entire dada2/decontam process — dada2_decontam_wrapper • micro4R</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Wrapper of entire dada2/decontam process — dada2_decontam_wrapper"><meta name="description" content="Wrapper of entire dada2/decontam process"><meta property="og:description" content="Wrapper of entire dada2/decontam process"><meta property="og:image" content="https://mshilts1.github.io/micro4R/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">micro4R</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Wrapper of entire dada2/decontam process</h1>

      <div class="d-none name"><code>dada2_decontam_wrapper.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Wrapper of entire dada2/decontam process</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">dada2_decontam_wrapper</span><span class="op">(</span><span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Arguments to pass to nested functions</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list of the new ASV table, taxa table, and metadata object</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">dada2_decontam_wrapper</span><span class="op">(</span>example <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Because you're running the example, any output files will go to a temporary directory, /tmp/RtmpoDrnyu/dada2_out. To avoid cluttering your computer, this folder and its contents should all be deleted at the end of your R session."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Because you're running the example, any output files will go to a temporary directory, /tmp/RtmpoDrnyu/dada2_out. To avoid cluttering your computer, this folder and its contents should all be deleted at the end of your R session."</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_307-ATAGTACC-ACGTCTCG_S307_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_307-ATAGTACC-ACGTCTCG_S307_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 11 (22%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_313-GACATAGT-TCGACGAG_S313_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_313-GACATAGT-TCGACGAG_S313_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 26 (52%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_328-GATCTACG-TCGACGAG_S328_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_328-GATCTACG-TCGACGAG_S328_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 46 (92%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_339-ACTCACTG-GATCGTGT_S339_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_339-ACTCACTG-GATCGTGT_S339_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 41 (82%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_162-ACGTGCGC-GGATATCT_S162_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_162-ACGTGCGC-GGATATCT_S162_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 37 (74%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_297-GTCTGCTA-ACGTCTCG_S297_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_297-GTCTGCTA-ACGTCTCG_S297_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 44 (88%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_381-TGCTCGTA-GTCAGATA_S381_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_381-TGCTCGTA-GTCAGATA_S381_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 43 (86%) filtered paired-sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59520 total bases in 248 reads from 7 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 .......</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  2  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49600 total bases in 248 reads from 7 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 .......</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  2  rounds.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_307-ATAGTACC-ACGTCTCG_S307_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 11 unique sequences from 11 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_313-GACATAGT-TCGACGAG_S313_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 25 unique sequences from 26 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_328-GATCTACG-TCGACGAG_S328_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 37 unique sequences from 46 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_339-ACTCACTG-GATCGTGT_S339_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 36 unique sequences from 41 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_162-ACGTGCGC-GGATATCT_S162_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 30 unique sequences from 37 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_297-GTCTGCTA-ACGTCTCG_S297_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 33 unique sequences from 44 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_381-TGCTCGTA-GTCAGATA_S381_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 36 unique sequences from 43 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_307-ATAGTACC-ACGTCTCG_S307_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 11 unique sequences from 11 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_313-GACATAGT-TCGACGAG_S313_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 23 unique sequences from 26 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_328-GATCTACG-TCGACGAG_S328_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 35 unique sequences from 46 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_339-ACTCACTG-GATCGTGT_S339_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 35 unique sequences from 41 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_162-ACGTGCGC-GGATATCT_S162_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 31 unique sequences from 37 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_297-GTCTGCTA-ACGTCTCG_S297_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 32 unique sequences from 44 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_381-TGCTCGTA-GTCAGATA_S381_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 37 unique sequences from 43 total sequences read.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 11 reads in 11 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 26 reads in 25 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 3 - 46 reads in 37 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 4 - 41 reads in 36 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 5 - 37 reads in 30 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 6 - 44 reads in 33 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 7 - 43 reads in 36 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 11 reads in 11 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 26 reads in 23 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 3 - 46 reads in 35 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 4 - 41 reads in 35 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 5 - 37 reads in 31 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 6 - 44 reads in 32 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 7 - 43 reads in 37 unique sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0 paired-reads (in 0 unique pairings) successfully merged out of 1 (in 1 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0 paired-reads (in 0 unique pairings) successfully merged out of 10 (in 1 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 44 paired-reads (in 1 unique pairings) successfully merged out of 44 (in 1 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 37 paired-reads (in 2 unique pairings) successfully merged out of 37 (in 2 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 28 paired-reads (in 3 unique pairings) successfully merged out of 28 (in 3 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 41 paired-reads (in 2 unique pairings) successfully merged out of 41 (in 2 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0 paired-reads (in 0 unique pairings) successfully merged out of 31 (in 3 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Identified 0 bimeras out of 6 input sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Because you're running the example, any output files will go to a temporary directory, /tmp/RtmpoDrnyu/dada2_out. To avoid cluttering your computer, this folder and its contents should all be deleted at the end of your R session."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Because you're running the example, any output files will go to a temporary directory, /tmp/RtmpoDrnyu/dada2_out. To avoid cluttering your computer, this folder and its contents should all be deleted at the end of your R session."</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_307-ATAGTACC-ACGTCTCG_S307_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_307-ATAGTACC-ACGTCTCG_S307_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 11 (22%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_313-GACATAGT-TCGACGAG_S313_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_313-GACATAGT-TCGACGAG_S313_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 26 (52%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_328-GATCTACG-TCGACGAG_S328_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_328-GATCTACG-TCGACGAG_S328_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 46 (92%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_339-ACTCACTG-GATCGTGT_S339_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_339-ACTCACTG-GATCGTGT_S339_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 41 (82%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_162-ACGTGCGC-GGATATCT_S162_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_162-ACGTGCGC-GGATATCT_S162_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 37 (74%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_297-GTCTGCTA-ACGTCTCG_S297_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_297-GTCTGCTA-ACGTCTCG_S297_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 44 (88%) filtered paired-sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_381-TGCTCGTA-GTCAGATA_S381_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting file:/tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_381-TGCTCGTA-GTCAGATA_S381_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Read in 50 paired-sequences, output 43 (86%) filtered paired-sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59520 total bases in 248 reads from 7 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 .......</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  2  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49600 total bases in 248 reads from 7 samples will be used for learning the error rates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 .......</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  2  rounds.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_307-ATAGTACC-ACGTCTCG_S307_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 11 unique sequences from 11 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_313-GACATAGT-TCGACGAG_S313_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 25 unique sequences from 26 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_328-GATCTACG-TCGACGAG_S328_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 37 unique sequences from 46 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_339-ACTCACTG-GATCGTGT_S339_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 36 unique sequences from 41 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_162-ACGTGCGC-GGATATCT_S162_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 30 unique sequences from 37 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_297-GTCTGCTA-ACGTCTCG_S297_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 33 unique sequences from 44 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_381-TGCTCGTA-GTCAGATA_S381_L001_F_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 36 unique sequences from 43 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_307-ATAGTACC-ACGTCTCG_S307_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 11 unique sequences from 11 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_313-GACATAGT-TCGACGAG_S313_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 23 unique sequences from 26 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_328-GATCTACG-TCGACGAG_S328_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 35 unique sequences from 46 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5080-MS-1_339-ACTCACTG-GATCGTGT_S339_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 35 unique sequences from 41 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_162-ACGTGCGC-GGATATCT_S162_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 31 unique sequences from 37 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_297-GTCTGCTA-ACGTCTCG_S297_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 32 unique sequences from 44 total sequences read.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dereplicating sequence entries in Fastq file: /tmp/RtmpoDrnyu/dada2_out/filtered/SAMPLED_5348-MS-1_381-TGCTCGTA-GTCAGATA_S381_L001_R_filt.fastq.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encountered 37 unique sequences from 43 total sequences read.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 11 reads in 11 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 26 reads in 25 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 3 - 46 reads in 37 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 4 - 41 reads in 36 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 5 - 37 reads in 30 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 6 - 44 reads in 33 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 7 - 43 reads in 36 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 1 - 11 reads in 11 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 2 - 26 reads in 23 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 3 - 46 reads in 35 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 4 - 41 reads in 35 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 5 - 37 reads in 31 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 6 - 44 reads in 32 unique sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sample 7 - 43 reads in 37 unique sequences.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0 paired-reads (in 0 unique pairings) successfully merged out of 1 (in 1 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0 paired-reads (in 0 unique pairings) successfully merged out of 10 (in 1 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 44 paired-reads (in 1 unique pairings) successfully merged out of 44 (in 1 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 37 paired-reads (in 2 unique pairings) successfully merged out of 37 (in 2 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 28 paired-reads (in 3 unique pairings) successfully merged out of 28 (in 3 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 41 paired-reads (in 2 unique pairings) successfully merged out of 41 (in 2 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0 paired-reads (in 0 unique pairings) successfully merged out of 31 (in 3 pairings) input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Identified 0 bimeras out of 6 input sequences.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "CAUTION: You're using the provided micro4R EXAMPLE reference databases. These are extremely tiny and unrealistic and meant only for testing and demonstration purposes. DO NOT use them with your real data."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Finished processing reference fasta.function (topic, package = NULL, lib.loc = NULL, character.only = FALSE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     give.lines = FALSE, local = FALSE, type = c("console", "html"), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     echo = TRUE, verbose = getOption("verbose"), setRNG = FALSE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ask = getOption("example.ask"), prompt.prefix = abbreviate(topic, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         6), catch.aborts = FALSE, run.dontrun = FALSE, run.donttest = interactive()) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type &lt;- match.arg(type)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     html &lt;- type == "html"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (html) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         enhancedHTML &lt;- str2logical(Sys.getenv("_R_HELP_ENABLE_ENHANCED_HTML_", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             "TRUE"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (!interactive() || !enhancedHTML || !requireNamespace("knitr", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             quietly = TRUE)) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             html &lt;- FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (html) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         port &lt;- tools::startDynamicHelp(NA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (port &lt;= 0L) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             html &lt;- FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             if (!is.null(lib.loc)) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 lib.loc &lt;- NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             browser &lt;- if (.Platform$GUI == "AQUA") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 get("aqua.browser", envir = as.environment("tools:RGUI"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             else getOption("browser")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!character.only) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         topic &lt;- substitute(topic)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (!is.character(topic)) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             topic &lt;- deparse(topic)[1L]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     pkgpaths &lt;- find.package(package, lib.loc, verbose = verbose)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     file &lt;- index.search(topic, pkgpaths, firstOnly = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!length(file)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         warning(gettextf("no help found for %s", sQuote(topic)), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             domain = NA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         return(invisible())</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (verbose) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cat("Found file =", sQuote(file), "\n")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     packagePath &lt;- dirname(dirname(file))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     pkgname &lt;- basename(packagePath)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (html) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         query &lt;- if (local) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ""</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         else "?local=FALSE"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         browseURL(paste0("http://127.0.0.1:", port, "/library/", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             pkgname, "/Example/", topic, query), browser)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         return(invisible())</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     lib &lt;- dirname(packagePath)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     tf &lt;- tempfile("Rex")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     tools::Rd2ex(.getHelpFile(file), tf, commentDontrun = !run.dontrun, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         commentDonttest = !run.donttest)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!file.exists(tf)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (give.lines) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             return(character())</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         warning(gettextf("%s has a help file but no examples", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             sQuote(topic)), domain = NA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         return(invisible())</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     on.exit(unlink(tf))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (give.lines) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         return(readLines(tf))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (pkgname != "base") </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         library(pkgname, lib.loc = lib, character.only = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!is.logical(setRNG) || setRNG) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if ((exists(".Random.seed", envir = .GlobalEnv))) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             oldSeed &lt;- get(".Random.seed", envir = .GlobalEnv)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             on.exit(assign(".Random.seed", oldSeed, envir = .GlobalEnv), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 add = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             oldRNG &lt;- RNGkind()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             on.exit(RNGkind(oldRNG[1L], oldRNG[2L], oldRNG[3L]), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 add = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (is.logical(setRNG)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             RNGkind("default", "default", "default")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             set.seed(1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         else eval(setRNG)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     zz &lt;- readLines(tf, n = 1L)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     skips &lt;- 0L</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (echo) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         zcon &lt;- file(tf, open = "rt")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         while (length(zz) &amp;&amp; !length(grep("^### \\*\\*", zz))) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             skips &lt;- skips + 1L</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             zz &lt;- readLines(zcon, n = 1L)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         close(zcon)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (ask == "default") </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ask &lt;- echo &amp;&amp; grDevices::dev.interactive(orNone = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (ask) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         oldask &lt;- if (.Device != "null device") </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             grDevices::devAskNewPage(ask = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         else getOption("device.ask.default", FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         on.exit(if (.Device != "null device") grDevices::devAskNewPage(oldask), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             add = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         op &lt;- options(device.ask.default = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         on.exit(options(op), add = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source(tf, local, echo = echo, prompt.echo = paste0(prompt.prefix, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         getOption("prompt")), continue.echo = paste0(prompt.prefix, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         getOption("continue")), verbose = verbose, max.deparse.length = Inf, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         encoding = "UTF-8", catch.aborts = catch.aborts, skip.echo = skips, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         keep.source = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x55ee2d03a100&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:utils&gt;</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in access(object, "sam_data", errorIfNULL):</span> sam_data slot is empty.</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Meghan Shilts.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

