# Package index

## All functions

- [`allOrients()`](https://mshilts1.github.io/micro4R/reference/allOrients.md)
  : all possible orientations of a DNA string Code was pulled from dada2
  document https://benjjneb.github.io/dada2/ITS_workflow.html Benjamin
  Callahan gets all the credit
- [`assess_example`](https://mshilts1.github.io/micro4R/reference/assess_example.md)
  : Larger example dataset for further runs
- [`assess_run()`](https://mshilts1.github.io/micro4R/reference/assess_run.md)
  : Assess sequencing results
- [`calcAlpha()`](https://mshilts1.github.io/micro4R/reference/calcAlpha.md)
  : Calculate standard alpha diversity indices
- [`calcBeta()`](https://mshilts1.github.io/micro4R/reference/calcBeta.md)
  : Calculate beta diversity dissimilarities with or without rarefaction
- [`checkAll()`](https://mshilts1.github.io/micro4R/reference/checkAll.md)
  : checker utils in this section Check your ASV count and taxonomy
  tables, and optionall metadata, for potential issues.
- [`checkMeta()`](https://mshilts1.github.io/micro4R/reference/checkMeta.md)
  : Check metadata file to identify potential downstream issues.
  DEPRECATED
- [`checkSampleID()`](https://mshilts1.github.io/micro4R/reference/checkSampleID.md)
  : Simple function to check for a column called SampleID
- [`contaminate()`](https://mshilts1.github.io/micro4R/reference/contaminate.md)
  : "Contaminate" the example ASV table so decontam can run in example
- [`converter()`](https://mshilts1.github.io/micro4R/reference/converter.md)
  : Converts tibbles, data frames, and matrices to each other as needed
  by the specific function
- [`cutadapt_helper()`](https://mshilts1.github.io/micro4R/reference/cutadapt_helper.md)
  : Generate a shell script to run cutadapt to remove primer sequences
- [`dada2_asvtable()`](https://mshilts1.github.io/micro4R/reference/dada2_asvtable.md)
  : dada2 workflow to generate ASV table
- [`dada2_decontam_wrapper()`](https://mshilts1.github.io/micro4R/reference/dada2_decontam_wrapper.md)
  : Wrapper of entire dada2/decontam process
- [`dada2_taxa()`](https://mshilts1.github.io/micro4R/reference/dada2_taxa.md)
  : dada2 taxa
- [`dada2_wrapper()`](https://mshilts1.github.io/micro4R/reference/dada2_wrapper.md)
  : dada2 wrapper
- [`decontam_wrapper()`](https://mshilts1.github.io/micro4R/reference/decontam_wrapper.md)
  : Decontam
- [`exampleData`](https://mshilts1.github.io/micro4R/reference/exampleData.md)
  : List of all three example objects in one package
- [`example_metadata()`](https://mshilts1.github.io/micro4R/reference/example_metadata.md)
  : Load in example metadata file
- [`filtering()`](https://mshilts1.github.io/micro4R/reference/filtering.md)
  : Filter the ASV table
- [`known_dada2_dbs()`](https://mshilts1.github.io/micro4R/reference/known_dada2_dbs.md)
  : A simple tibble of the names of maintained dada2-formatted taxonomy
  reference databases
- [`makeExample()`](https://mshilts1.github.io/micro4R/reference/makeExample.md)
  : Create example data for testing and teaching purposes specifically
  for this package.
- [`makeExampleMeta()`](https://mshilts1.github.io/micro4R/reference/makeExampleMeta.md)
  : Create a simple metadata table
- [`makeExampleSeqtab()`](https://mshilts1.github.io/micro4R/reference/makeExampleSeqtab.md)
  : Creates a simple example ASV table
- [`makeExampleTaxa()`](https://mshilts1.github.io/micro4R/reference/makeExampleTaxa.md)
  : Create a simple taxonomy table
- [`packItUp()`](https://mshilts1.github.io/micro4R/reference/packItUp.md)
  : Package up your three objects (metadata, asvtable, taxonomy table)
  into a single list for less typing
- [`primerHits()`](https://mshilts1.github.io/micro4R/reference/primerHits.md)
  : primer hits Code was pulled from dada2 document
  https://benjjneb.github.io/dada2/ITS_workflow.html Benjamin Callahan
  gets all the credit
- [`ref_db()`](https://mshilts1.github.io/micro4R/reference/ref_db.md) :
  Set a reference database for taxonomy assignment
- [`test_dbs()`](https://mshilts1.github.io/micro4R/reference/test_dbs.md)
  : Load test databases into environment without having to type so much.
  Because I'm lazy and got really tired of finding the exact text to
  copy paste
- [`tibblefy()`](https://mshilts1.github.io/micro4R/reference/tibblefy.md)
  : tibblefy a specific type of data frame. DEPRECATED. use converter()
  instead
- [`unsampled_example()`](https://mshilts1.github.io/micro4R/reference/unsampled_example.md)
  : Load in unsampled example
- [`whereFastqs()`](https://mshilts1.github.io/micro4R/reference/whereFastqs.md)
  : Collection of helper/utility functions not really intended to be
  directly used by end user Path to user's fastq file folder
